src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -86,6 +86,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <NuGetAuditMode Condition=" '$(NuGetAuditMode)' == '' ">all</NuGetAuditMode>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <RestoreEnablePackagePruning Condition="$(RestoreEnablePackagePruning) == '' ">false</RestoreEnablePackagePruning>
```

Comment:
I go back and forth on this, but its my personal opinion that we should never declare a property for a boolean that we control the default value of.  Basically, if its declared by a user that's fine, but if its not declared then we respect a default.  Every property that's declared is just one more string that every project will allocate.  If this property is going to default to false for now and later on we'll default it to true, then we can remove the property declaration here.  Just my two cents.

Reply:
We'll make it true eventually. I had it true in early parts of my implementation. 

We'll definitely make it key off SDKAnalysisLevel and TFM targetted (enabled for .NET 10 for example).

Reply:
I know that declaring is required for the project-system properties page, but this won't really go there.